plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/AccessTrackingFileSystem.java

```diff
@@ -108,20 +349,623 @@ public Path getWorkingDirectory()
+    {
+        return fileSystemDelegate.createSnapshot(path, snapshotName);
+    }
+
+    @Override
+    public void renameSnapshot(Path path, String snapshotOldName, String snapshotNewName)
+            throws IOException
+    {
+        fileSystemDelegate.renameSnapshot(path, snapshotOldName, snapshotNewName);
+    }
+
+    @Override
+    public void deleteSnapshot(Path path, String snapshotName)
+            throws IOException
+    {
+        fileSystemDelegate.deleteSnapshot(path, snapshotName);
+    }
+
+    @Override
+    public void modifyAclEntries(Path path, List<AclEntry> aclSpec)
+            throws IOException
+    {
+        fileSystemDelegate.modifyAclEntries(path, aclSpec);
+    }
+
+    @Override
+    public void removeAclEntries(Path path, List<AclEntry> aclSpec)
+            throws IOException
+    {
+        fileSystemDelegate.removeAclEntries(path, aclSpec);
+    }
+
+    @Override
+    public void removeDefaultAcl(Path path)
+            throws IOException
+    {
+        fileSystemDelegate.removeDefaultAcl(path);
+    }
+
+    @Override
+    public AclStatus getAclStatus(Path path)
+            throws IOException
+    {
+        return fileSystemDelegate.getAclStatus(path);
+    }
+
+    @Override
+    public void setStoragePolicy(Path src, String policyName)
+            throws IOException
+    {
+        fileSystemDelegate.setStoragePolicy(src, policyName);
+    }
+
+    @Override
+    public void unsetStoragePolicy(Path src)
+            throws IOException
+    {
+        fileSystemDelegate.unsetStoragePolicy(src);
+    }
+
+    @Override
+    public BlockStoragePolicySpi getStoragePolicy(Path src)
+            throws IOException
+    {
+        return fileSystemDelegate.getStoragePolicy(src);
+    }
+
+    @Override
+    public Collection<? extends BlockStoragePolicySpi> getAllStoragePolicies()
+            throws IOException
+    {
+        return fileSystemDelegate.getAllStoragePolicies();
+    }
+
+    @Override
+    public Path getTrashRoot(Path path)
+    {
+        return fileSystemDelegate.getTrashRoot(path);
+    }
+
+    @Override
+    public Collection<FileStatus> getTrashRoots(boolean allUsers)
+    {
+        return fileSystemDelegate.getTrashRoots(allUsers);
+    }
+
+    @Override
+    public StorageStatistics getStorageStatistics()
+    {
+        return fileSystemDelegate.getStorageStatistics();
+    }
+
+    @Override
+    public FSDataOutputStreamBuilder createFile(Path path)
+    {
+        return fileSystemDelegate.createFile(path);
+    }
+
+    @Override
+    public Configuration getConf()
+    {
+        return fileSystemDelegate.getConf();
+    }
+
+    @Override
+    public void setConf(Configuration conf)
+    {
+        // fileSystemDelegate is not yet initialized when this method is called from the default constructor of FileSystem
+        if (fileSystemDelegate == null) {
+            super.setConf(conf);
+        }
+        else {
+            fileSystemDelegate.setConf(conf);
+        }
```

Comment:
@findepi Modified from your last review because `TestTransactionLogAccess` failed due to NPE of `fileSystemDelegate`. Could you please take another look when you have time?